provide/Services/NatsClient.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NATS.Client;
+
+namespace provide 
+{
+    public class NatsClient 
+    {
+        private string host;
+        private string path;
+        private string scheme;
+        private string token;
+        private readonly string natsUrl;
+        private IConnection connection;
+        private Dictionary<string, IAsyncSubscription> subscriptions = new Dictionary<string, IAsyncSubscription>();
+
+        public NatsClient(string token)
+        {
+            this.token = token;
+        }
+
+        public NatsClient(string host, string path, string scheme, string token)
+        {
+            this.host = host;
+            this.path = path;
+            this.scheme = scheme;
+            this.token = token;
+            this.natsUrl = $"{scheme}://${host}/${path}/";
+        }
+
+        public void Connect()
+        {
+            var opts = ConnectionFactory.GetDefaultOptions();
+            SetJWTAndNKeyHandlers(opts);
+            // if not provided it will use default local server url
+            opts.Url = this.natsUrl;
+            // add try catch?
+            this.connection = new ConnectionFactory().CreateConnection(opts);
+        }
+
+        public ConnState GetConnectionState()
+        {
+            return this.connection.State;
+        }
+
+        public void Close()
+        {
+            if (this.connection.IsClosed()) 
+            {
+                return;
+            }
+            // should we do drain instead?
+            this.connection.Close();
```

Comment:
Does this automatically close existing subscriptions? If it works how other NATS clients do, I would assume not.

Reply:
yes, from docs:   Closes the NATS.Client.IConnection and all associated subscriptions.